src/consensus/vote.cpp

```diff
@@ -549,6 +564,106 @@ std::string VoteManager::getJsonStr(std::vector<Vote> const& votes) {
   return ptroot.toStyledString();
 }
 
+std::vector<Vote> VoteManager::getProposalVotes(uint64_t const pbft_round) {
+  std::unordered_map<blk_hash_t, std::unordered_set<vote_hash_t>> proposal_voted_value_map;
+  {
+    sharedLock_ lock(verified_votes_access_);
+    proposal_voted_value_map = verified_votes_[pbft_round][1];
+  }
+
+  std::vector<Vote> proposal_votes;
+  for (auto const& voted_value : proposal_voted_value_map) {
+    for (auto const& v_hash : voted_value.second) {
+      auto vote = db_->getVerifiedVote(v_hash);
+      if (vote) {
+        proposal_votes.emplace_back(*vote);
```

Comment:
why would copy votes here ? Is there some reason why not to return std::vector<std::shared_ptr<Vote>> ???

Reply:
fixed, thanks